src/basic/architecture.h

```diff
@@ -45,7 +45,7 @@ enum {
         ARCHITECTURE_X86_64,
         _ARCHITECTURE_MAX,
         _ARCHITECTURE_INVALID = -EINVAL,
-};
+} Architecture;
```

Comment:
what's the rationale here? So far we avoided this since we wanted to be able to return the full range of libc errors from the architecture functions if needed, and thus we pretty much need to use 'int' there, and not an enum...

Reply:
We're storing the arch in GptPartitionType in a later commit and there it makes sense to limit the range of vaIues think.